fathom/src/core/binary.rs

```diff
@@ -183,7 +184,33 @@ impl<'arena, 'env> Context<'arena, 'env> {
         Ok(Arc::new(Value::ArrayLit(elem_exprs)))
     }
 
-    pub fn read_link(
+    fn read_repeat_until_end(
+        &mut self,
+        reader: &mut dyn SeekRead,
+        elem_format: &ArcValue<'arena>,
+    ) -> Result<ArcValue<'arena>, io::Error> {
+        let mut current_pos = reader.stream_position()?;
+        let mut elems = Vec::new();
+
+        loop {
+            match self.read_format(reader, elem_format) {
+                Ok(elem) => {
+                    elems.push(elem);
+                    current_pos = reader.stream_position()?;
+                }
+                Err(err) => match err.kind() {
+                    io::ErrorKind::UnexpectedEof => {
+                        // FIXME: should this be set to the end of the current stream?
+                        reader.seek(SeekFrom::Start(current_pos))?;
+                        return Ok(Arc::new(Value::ArrayLit(elems)));
+                    }
+                    _ => return Err(err),
```

Comment:
Does this allow trailing random bytes at the end of the stream?

Reply:
Ahh yeah it does - I was thinking of maybe adding a `repeat_until_end_exact` format.